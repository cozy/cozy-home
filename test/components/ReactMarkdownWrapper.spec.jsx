import React from 'react'
import { render } from '@testing-library/react'

import {
  ReactMarkdownWrapper,
  reactMarkdownRendererOptions
} from 'components/ReactMarkdownWrapper'

describe('ReactMarkdownWrapper component', () => {
  beforeEach(() => {
    jest.resetModules()
  })

  it('should be displayed correctly if source provided', () => {
    const { container } = render(
      <ReactMarkdownWrapper source={'**test** using `markdown`'} />
    )
    expect(container).toMatchSnapshot()
  })

  it('should be displayed correctly if source with link', () => {
    const { container } = render(
      <ReactMarkdownWrapper
        source={
          '**test** using [markdown](https://en.wikipedia.org/wiki/Markdown)'
        }
      />
    )
    expect(container).toMatchSnapshot()
  })
})

describe('ReactMarkdown options', () => {
  it('should correctly return an anchor HTML element', () => {
    expect(
      reactMarkdownRendererOptions.Link({
        href: 'https://testlink.mock',
        children: 'example link test for test'
      })
    ).toMatchSnapshot()
  })
})
