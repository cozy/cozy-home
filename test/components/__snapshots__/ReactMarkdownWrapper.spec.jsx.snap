// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactMarkdown options should correctly return an anchor HTML element 1`] = `
<a
  href="https://testlink.mock"
  rel="noopener noreferrer"
  target="_blank"
>
  example link test for test
</a>
`;

exports[`ReactMarkdownWrapper component should be displayed correctly if source provided 1`] = `
<div>
  <p>
    <strong>
      test
    </strong>
     using 
    <code>
      markdown
    </code>
  </p>
</div>
`;

exports[`ReactMarkdownWrapper component should be displayed correctly if source with link 1`] = `
<div>
  <p>
    <strong>
      test
    </strong>
     using 
    <a
      href="https://en.wikipedia.org/wiki/Markdown"
    >
      markdown
    </a>
  </p>
</div>
`;
